solidity/test/data/fraud.ts

```diff
@@ -51,15 +86,16 @@ export const nonWitnessSignSingleInputTx: FraudTestData = {
     "a90f738986f60000000001000000",
   sighash: "0x5d09cd07392c7163335b67eacc999491a3794c15b88e2b59094be5c5b157064b",
   witness: false,
-  deposits: [
+  deposits: [],
+  spentMainUtxos: [],
+  movedFundsSweepRequests: [
```

Comment:
Non-blocking but I think we should improve unit tests. Right now, the `self.movedFundsSweepRequests[utxoKey].state == MovingFunds.MovedFundsSweepRequestState.Processed` is covered kind of "accidentally", because we register `nonWitnessSignSingleInputTx` case as `movedFundsSweepRequests`.

I think we should distinguish in unit tests swept deposits UTXOs, spent main UTXOs, and moved funds sweep requests and have those three cases tested explicitly for all types of transactions (witness, non-witness, single input, multiple inputs).